packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -752,4 +754,150 @@ describe("<FilterBar />", () => {
       })
     })
   })
+
+  describe("Context use cases", () => {
+    type Items = Array<{ value: string; label: string }>
+
+    type AsyncValues = {
+      city: string[]
+      hero: string[]
+    }
+
+    const MockFilterAsyncComponent = ({
+      id,
+      fetcher,
+    }: {
+      id: string
+      fetcher: (args: Partial<AsyncValues>) => Promise<Items>
+    }): JSX.Element => {
+      const [items, setItems] = useState<Items>([])
+      const { getActiveFilterValues } = useFilterBarContext()
+      const activeFilterVals = getActiveFilterValues()
+
+      useEffect(() => {
+        fetcher(activeFilterVals).then(fetchedItems => {
+          if (JSON.stringify(fetchedItems) !== JSON.stringify(items)) {
+            setItems(fetchedItems)
+          }
+        })
+      }, [JSON.stringify(activeFilterVals)])
+
+      return (
+        <FilterBar.MultiSelect id={id} items={items}>
+          {() => (
+            <FilterMultiSelect.ListBox>
+              {({ allItems }) => (
+                <FilterMultiSelect.ListBoxSection
+                  items={allItems}
+                  sectionName="All Items"
+                >
+                  {item => (
+                    <FilterMultiSelect.Option key={item.key} item={item} />
+                  )}
+                </FilterMultiSelect.ListBoxSection>
+              )}
+            </FilterMultiSelect.ListBox>
+          )}
+        </FilterBar.MultiSelect>
+      )
+    }
+
+    const fetchCityOptions = jest.fn((filterValues: Partial<AsyncValues>) => {
+      const isSupermanInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("3") > -1
+      )
+      const isBatmanInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("4") > -1
+      )
+
+      if (isBatmanInFilterValue && !isSupermanInFilterValue) {
+        return Promise.resolve([{ value: "1", label: "Gotham" }])
+      }
+
+      return Promise.resolve([
+        { value: "1", label: "Gotham" },
+        { value: "2", label: "Metropolis" },
+      ])
```

Comment:
It will be clearer if we have the `value`s be strings as opposed to numbers (eg. `value: "gotham", label: "Gotham"`).

Reply:
Sounds good. I was thinking this too, but got lazy. 😩